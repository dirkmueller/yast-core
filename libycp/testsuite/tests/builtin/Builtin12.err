Parsed:
----------------------------------------------------------------------
{
    // list SetOne
    // list SetEqual
    // list SetTwo
    // list SetDifferent
    // list ResultSet
    // filename: "tests/builtin/Builtin12.ycp"
    list SetOne = [0, "Hello world", true];
    list SetEqual = [select (SetOne, 2, nil), (7 * 0), ("Hello " + "world")];
    list SetTwo = [., (15 % 5)];
    list SetDifferent = [3.];
    list ResultSet = [];
    ResultSet = union (SetOne, SetOne);
    return (ResultSet == SetOne);
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
    // list SetOne
    // list SetEqual
    // list SetTwo
    // list SetDifferent
    // list ResultSet
    // filename: "tests/builtin/Builtin12.ycp"
    list SetOne = [0, "Hello world", true];
    list SetEqual = [select (SetOne, 2, nil), (7 * 0), ("Hello " + "world")];
    list SetTwo = [., (15 % 5)];
    list SetDifferent = [3.];
    list ResultSet = [];
    ResultSet = union (SetOne, SetEqual);
    return ((ResultSet == SetOne) && (ResultSet != SetEqual));
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
    // list SetOne
    // list SetEqual
    // list SetTwo
    // list SetDifferent
    // list ResultSet
    // filename: "tests/builtin/Builtin12.ycp"
    list SetOne = [0, "Hello world", true];
    list SetEqual = [select (SetOne, 2, nil), (7 * 0), ("Hello " + "world")];
    list SetTwo = [., (15 % 5)];
    list SetDifferent = [3.];
    list ResultSet = [];
    ResultSet = union (SetOne, SetTwo);
    return (ResultSet == add (SetOne, select (SetTwo, 0, nil)));
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
    // list SetOne
    // list SetEqual
    // list SetTwo
    // list SetDifferent
    // list ResultSet
    // filename: "tests/builtin/Builtin12.ycp"
    list SetOne = [0, "Hello world", true];
    list SetEqual = [select (SetOne, 2, nil), (7 * 0), ("Hello " + "world")];
    list SetTwo = [., (15 % 5)];
    list SetDifferent = [3.];
    list ResultSet = [];
    ResultSet = union (SetOne, SetDifferent);
    return (ResultSet == add (SetOne, select (SetDifferent, 0, nil)));
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
    // list SetOne
    // list SetEqual
    // list SetTwo
    // list SetDifferent
    // list ResultSet
    // filename: "tests/builtin/Builtin12.ycp"
    list SetOne = [0, "Hello world", true];
    list SetEqual = [select (SetOne, 2, nil), (7 * 0), ("Hello " + "world")];
    list SetTwo = [., (15 % 5)];
    list SetDifferent = [3.];
    list ResultSet = [];
    ResultSet = union (SetDifferent, SetOne);
    return (ResultSet != add (SetOne, select (SetDifferent, 0, nil)));
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
    // filename: "tests/builtin/Builtin12.ycp"
    return "test case union finished";
}
----------------------------------------------------------------------
