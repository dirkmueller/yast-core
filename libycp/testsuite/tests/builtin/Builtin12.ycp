
# ---------------------------------------------------------
#
#  Filename:    Builtin12.ycp
#
#  Purpose:     test cases for YCP test
#
#  Creator:     jd@suse.de
#
#  Maintainer:  jd@suse.de
#
# ---------------------------------------------------------

// Interprets two lists as sets and returns a new list that has
// all elements of the first list and all of the second list. Identical
// elements are dropped. The order of the elements in the new list is
// preserved. Elements of the first list are prior to elements from 
// the second list.

{
    list   SetOne = [ 0 , "Hello world", true ];
    list   SetEqual = [ select( SetOne, 2, nil), 7 * 0, "Hello " + "world" ];
    list   SetTwo = [ ., 15 % 5 ];
    list   SetDifferent = [ 3.00 ];
    list   ResultSet = [];

    ResultSet = union( SetOne, SetOne );
    return( ResultSet == SetOne );
}

{
    list   SetOne = [ 0 , "Hello world", true ];
    list   SetEqual = [ select( SetOne, 2, nil), 7 * 0, "Hello " + "world" ];
    list   SetTwo = [ ., 15 % 5 ];
    list   SetDifferent = [ 3.00 ];
    list   ResultSet = [];

    ResultSet = union( SetOne, SetEqual );
    return( ( ResultSet == SetOne ) &&
            ( ResultSet != SetEqual ) );
}

{
    list   SetOne = [ 0 , "Hello world", true ];
    list   SetEqual = [ select( SetOne, 2, nil), 7 * 0, "Hello " + "world" ];
    list   SetTwo = [ ., 15 % 5 ];
    list   SetDifferent = [ 3.00 ];
    list   ResultSet = [];

    ResultSet = union( SetOne, SetTwo );
    return( ResultSet == add( SetOne, select( SetTwo, 0, nil) ) );
}

{
    list   SetOne = [ 0 , "Hello world", true ];
    list   SetEqual = [ select( SetOne, 2, nil), 7 * 0, "Hello " + "world" ];
    list   SetTwo = [ ., 15 % 5 ];
    list   SetDifferent = [ 3.00 ];
    list   ResultSet = [];

    ResultSet = union( SetOne, SetDifferent );
    return( ResultSet == add( SetOne, select( SetDifferent, 0, nil) ) );
}

{
    list   SetOne = [ 0 , "Hello world", true ];
    list   SetEqual = [ select( SetOne, 2, nil), 7 * 0, "Hello " + "world" ];
    list   SetTwo = [ ., 15 % 5 ];
    list   SetDifferent = [ 3.00 ];
    list   ResultSet = [];

    ResultSet = union( SetDifferent, SetOne );
    return( ResultSet != add( SetOne, select( SetDifferent, 0, nil) ) );
}

{
    return( "test case union finished" );
}


