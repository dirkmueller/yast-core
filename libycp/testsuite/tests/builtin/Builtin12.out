(true)
(true)
(true)
(true)
(true)
("test case union finished")
